### ISSUE TITLE
support prepare-commit-msg hook

### ISSUE THREAD

**Issue by extrawurst** [extrawurst/gitui#1873]
spun off of #1711.

@pm100 are you still going to provide a PR on this?

**Comment by pm100** [comment]
I have working code but  I am blocked trying to work out what to do with the hook sequencing (as raised in #1711). I cant say I like what you propose for that (there would be no way to prevent the pre-commit running for example) I see you said that its a low priority for you, but it would be nice to have hook parity with git cli

**Comment by extrawurst** [comment]
> I have working code but I am blocked trying to work out what to do with the hook sequencing

don't see why the order matters for `prepare-commit-msg` - it operates on the commit message (default). It simply runs when the commit popup opens. 

lets discuss your concerns in #1711

**Comment by extrawurst** [comment]
see https://github.com/dvershinin/CommitGPT/blob/main/prepare-commit-msg.sh for an example

**Comment by pm100** [comment]
I have code pretty much working but disappeared down the textinput rabbit hole for months. Spent a long long time getting features added to it so that it works nicely in gitui. I also worked out how to make it work by only changing one file in gitui. There are also real nice features that I got in coming in their next version , including full kb driven text select / copy / paste to and from the text area. Fixes your unicode bug(s) too. The only thing blocking it is the whole cargo update versioning thing: my PR ends up having bitflags 2 pulled in, plus a different version of syn.
